### DISCUSSION TITLE
create_async_engine - why asyncpg only?

### DISCUSSION THREAD

**Discussion by mgurg** [agronholm/apscheduler#804] [Q&A] [answered]
Is any particular reason why only `asyncpg` is supported in [from_async_sqla_engine method](https://github.com/agronholm/apscheduler/blob/a7c93dac4dad6fa3a5f1e37ce760d85846689f6f/src/apscheduler/eventbrokers/asyncpg.py#L86) ?

I' using `psycopg3` to do sync (migrations) / async (regular db queries) db interactions. 

I added `asyncpg` just to be able to use APS, I wonder if in the future I will be able to use psycopg everywhere.

**Comment by agronholm** [comment, marked as answer]
This is because the event broker relies on asyncpg's own API for listening to notifications. I am open to providing support for psycopg3 too (as that is what I myself use in production). A PR (with tests) would make this happen a lot faster.

**Reply by agronholm** [reply]
Looks like it should be pretty easy with this: https://www.psycopg.org/psycopg3/docs/api/connections.html#psycopg.AsyncConnection.notifies

**Reply by agronholm** [reply]
FYI, I've pushed a psycopg event broker to `master`. It will be in the next pre-release.
